### PR TITLE
fix: nightly auto-update detection ignoring pre-release versions

### PR DESCRIPTION
**Pull request by donovan-yohan**
## Summary

Users on the nightly update channel always saw "up to date" in settings even when newer nightly builds were published. The version comparison function stripped pre-release suffixes before comparing, so `0.1.0-nightly.20260404.200` and `0.1.0-nightly.20260405.238` both collapsed to `0.1.0` and compared as equal.

## Changes

- **server/utils.ts**: Rewrote `semverLessThan` to compare pre-release identifiers per semver spec (numeric segments numerically, string segments lexically, fewer identifiers = lower precedence)
- **server/index.ts**: Added nightly channel fallback for the stable-to-nightly edge case (user on `0.1.0` switches to nightly, registry returns `0.1.0-nightly.x`)
- **test/version.test.ts**: Replaced stale inline function copy with import from production code, added 9 nightly-specific test cases
- **test/hooks.test.ts**: Updated assertion that expected the old broken behavior

## Testing

- 41/41 tests pass (version + hooks test files)
- TypeScript type check clean
- Verified all scenarios: nightly-to-nightly, stable-to-nightly, same build, reverse direction, identifier count edge cases
- Confirmed against live npm registry: `relay-ide/nightly` returns `0.1.0-nightly.20260405.238`

---
*Created with `/pr:author`*

### CODE REVIEW

File: server/utils.ts [hunk 1]
```diff
@@ -7,19 +7,43 @@ export function stripAnsi(text: string): string {
...
-    bPat = pb[2] ?? 0;
-  if (aMaj !== bMaj) return aMaj < bMaj;
-  if (aMin !== bMin) return aMin < bMin;
-  return aPat < bPat;
+  const idxA = a.indexOf('-');
+  const idxB = b.indexOf('-');
+  const aCore = (idxA === -1 ? a : a.slice(0, idxA)).split('.').map(Number);
+  const bCore = (idxB === -1 ? b : b.slice(0, idxB)).split('.').map(Number);
+  const aPre = idxA === -1 ? undefined : a.slice(idxA + 1);
+  const bPre = idxB === -1 ? undefined : b.slice(idxB + 1);
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply server/utils.ts]: Fixed in 3df45a9 — now strips build metadata via `a.split("+", 1)[0]` before parsing. Added test coverage for build metadata scenarios.

File: server/utils.ts [hunk 2]
```diff
@@ -7,19 +7,43 @@ export function stripAnsi(text: string): string {
...
+    if (i >= aIds.length) return true; // fewer identifiers = lower
+    if (i >= bIds.length) return false;
+    const aNum = Number(aIds[i]);
+    const bNum = Number(bIds[i]);
+    const aIsNum = !isNaN(aNum);
+    const bIsNum = !isNaN(bNum);
+    if (aIsNum && bIsNum) {
+      if (aNum !== bNum) return aNum < bNum;
+    } else if (aIsNum !== bIsNum) {
+      return aIsNum; // numeric < string per semver
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply server/utils.ts]: Fixed in 3df45a9 — replaced `Number()`/`isNaN` with `/^\d+$/` regex for strict ASCII-digits-only detection per SemVer spec.

File: server/index.ts [hunk 3]
```diff
@@ -1987,7 +1987,13 @@ async function main(): Promise<void> {
...
     const channel = startupConfig.updateChannel ?? 'stable';
     const latest = await getLatestVersion(channel);
-    const updateAvailable = latest !== null && semverLessThan(current, latest);
+    const updateAvailable =
+      latest !== null &&
+      current !== latest &&
+      (semverLessThan(current, latest) ||
+        (channel === 'nightly' &&
+          !current.includes('-') &&
+          latest.includes('nightly')));
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply server/index.ts]: Fixed in 3df45a9 — added `latest.startsWith(current + "-")` guard so the nightly fallback only fires when the nightly build shares the same core version, preventing downgrade prompts.
